src/dataServices/dynamoDbHelper.ts

```diff
@@ -50,6 +50,19 @@ export default class DynamoDbHelper {
         return items;
     }
 
+    async getAllResources(
```

Comment:
This method assumes that resources will not have more than 1000 versions. I think this is a reasonable limit. Even with the auto-update scheme of Origin Desktop, the user would have to have the form open for 1000 versions * 5 seconds / version = 5000 seconds = 83.33 minutes. Plus the 5 second interval is only triggered when the user makes and change that isn't followed by another change for 5 seconds. 

Nonetheless, how did you come to 1000? Maybe we can replace this '1000' with a DEFAULT_MAX_VERSIONS variable with a commented rationale for why.